BUG: fix: `list` as index item does not raise (#61674)
pandas-dev/pandas: doc/source/whatsnew/v3.0.0.rst

- Bug in :meth:`MultiIndex.insert` when a new value inserted to a datetime-like level gets cast to ``NaT`` and fails indexing (:issue:`60388`)
- Bug in printing :attr:`Index.names` and :attr:`MultiIndex.levels` would not escape single quotes (:issue:`60190`)
- Bug in reindexing of :class:`DataFrame` with :class:`PeriodDtype` columns in case of consolidated block (:issue:`60980`, :issue:`60273`)

> - Bug in creating :class:`Index` with one item being a ``list`` among others that aren't (should raise ValueError) (:issue:`60925`)

Reviewer: ```suggestion
- Bug in creating :class:`Index` with some items being a non-hashable; this now raises a ``ValueError`` (:issue:`60925`)
```
